README.md

```diff
@@ -1 +1,178 @@
+
+[Verify Passkeys Overview](https://www.twilio.com/docs/verify/passkeys)
+
+## Requirements <a name="requirements"></a>
+
+* Node.js v20.x or higher
+
+## Installation <a name="installation"></a>
+
+* Add the library to your project
+
+### Using NPM
+
+```
+npm install @twilio/twilio-verify-passkeys-web
+```
+
+### Directly from Github
+
+```
+npm install https://github.com/twilio/twilio-verify-passkeys-web.git
+```
+
+## Quickstart <a name="quickstart"></a>
+
+### Create registration
+
+Use the `TwilioPasskeys` instance to create a registration by calling the `create(String)` function.
+
+The `create` method receive a param that represent a challenge payload, check how to [create challenge payload](#create-challenge-payload).
+
+You can also call `create(CreatePasskeysRequest)`, where the `CreatePasskeysRequest` is wrapper object of a [creation challenge payload](#create-challenge-payload) schema.
+
+```js
+const createPasskeysResult = twilioPasskeys.create(challengePayload)
+
+if(createPasskeysResult.Error !== null) {
+    // verify the createPasskeyResult.createPasskeyResponse against your backend and finish sign up
+} else {
+    // handle error
+}
+```
+
+### Authenticate a user
+
+Use the `TwilioPasskeys` instance to authenticate a user by calling the `authenticate(String)` function.
+
+The `authenticate` method receive a param that represent an authentication request, it follows the schema of an [authentication challenge payload](#authenticate-challenge-payload).
+
+You can also call `authenticate(AuthenticatePasskeysRequest)`, where the `AuthenticatePasskeysRequest` is a wrapper object of an [authentication challenge payload](#authenticate-challenge-payload).
+
+```js
+const authenticatePasskeyResult = twilioPasskeys.authenticate(authenticationRequest)
+
+if(authenticatePasskeyResult.Error !== null) {
+    // verify the authenticatePasskeyResult.authenticatePasskeyResponse against your backend
+} else {
+    // handle error
+}
+```
+
+### Create Challenge Payload <a name="create-challenge-payload"></a>
+
+The challenge payload for creating a registration is a String obtained by requesting your backend a challenge for registering a user, it uses the JSON schema:
+
+```json
+{"rp":{"id":"your_backend","name":"PasskeySample"},"user":{"id":"WUV...5Ng","name":"1234567890","displayName":"1234567890"},"challenge":"WUY...jZQ","pubKeyCredParams":[{"type":"public-key","alg":-7}],"timeout":600000,"excludeCredentials":[],"authenticatorSelection":{"authenticatorAttachment":"platform","requireResidentKey":false,"residentKey":"preferred","userVerification":"preferred"},"attestation":"none"}
+```
+
+### Authenticate Challenge Payload <a name="authenticate-challenge-payload"></a>
+
+The challenge payload for authenticating a user is a JSON with the schema:
+
+```json
+{"publicKey":{"challenge":"WUM...2Mw","timeout":300000,"rpId":"your_backend","allowCredentials":[],"userVerification":"preferred"}}
+```
+
+## Running Tests <a name="running-tests"></a>
```

Comment:
I would move this section to the end. We do not expect developers to run this part often, instead the instructions to run the sample app will be more used

Reply:
Done!